src/Model/CustomerTokenService.php

```diff
@@ -0,0 +1,144 @@
+
+namespace ScandiPWA\CustomerGraphQl\Model;
+
+use Magento\Customer\Api\AccountManagementInterface;
+use Magento\Framework\Exception\EmailNotConfirmedException;
+use Magento\Integration\Model\CredentialsValidator;
+use Magento\Integration\Model\Oauth\TokenFactory as TokenModelFactory;
+use Magento\Integration\Model\ResourceModel\Oauth\Token\CollectionFactory as TokenCollectionFactory;
+use Magento\Integration\Model\Oauth\Token\RequestThrottler;
+use Magento\Framework\Exception\AuthenticationException;
+use Magento\Framework\Event\ManagerInterface;
+use Magento\Integration\Model\CustomerTokenService as SourceCustomerTokenService;
+
+/**
+ * Class CustomerTokenService
+ * @package ScandiPWA\CustomerGraphQl\Model
+ */
+class CustomerTokenService extends SourceCustomerTokenService
+{
+    /**
+     * Token Model
+     *
+     * @var TokenModelFactory
+     */
+    private $tokenModelFactory;
+
+    /**
+     * @var Magento\Framework\Event\ManagerInterface
+     */
+    private $eventManager;
+
+    /**
+     * Customer Account Service
+     *
+     * @var AccountManagementInterface
+     */
+    private $accountManagement;
+
+    /**
+     * @var \Magento\Integration\Model\CredentialsValidator
+     */
+    private $validatorHelper;
+
+    /**
+     * Token Collection Factory
+     *
+     * @var TokenCollectionFactory
+     */
+    private $tokenModelCollectionFactory;
+
+    /**
+     * @var RequestThrottler
+     */
+    private $requestThrottler;
+
+    /**
+     * Initialize service
+     *
+     * @param TokenModelFactory $tokenModelFactory
+     * @param AccountManagementInterface $accountManagement
+     * @param TokenCollectionFactory $tokenModelCollectionFactory
+     * @param \Magento\Integration\Model\CredentialsValidator $validatorHelper
+     * @param \Magento\Framework\Event\ManagerInterface $eventManager
+     */
+    public function __construct(
+        TokenModelFactory $tokenModelFactory,
+        AccountManagementInterface $accountManagement,
+        TokenCollectionFactory $tokenModelCollectionFactory,
+        CredentialsValidator $validatorHelper,
+        ManagerInterface $eventManager = null
+    ) {
+        parent::__construct(
+            $tokenModelFactory,
+            $accountManagement,
+            $tokenModelCollectionFactory,
+            $validatorHelper,
+            $eventManager ?: \Magento\Framework\App\ObjectManager::getInstance()
+                ->get(ManagerInterface::class)
+        );
+
+        $this->tokenModelFactory = $tokenModelFactory;
+        $this->accountManagement = $accountManagement;
+        $this->validatorHelper = $validatorHelper;
+        $this->eventManager = $eventManager ?: \Magento\Framework\App\ObjectManager::getInstance()
+            ->get(ManagerInterface::class);
+    }
+
+    /**
+     * Create customer access token
+     * @param $username
+     * @param $password
+     * @return string
+     */
+    public function createCustomerAccessToken($username, $password)
```

Comment:
same on methods, if we can support 7.4+ php only, defining strict return type of the methods would be nice